hwcomposer/hwcomposer_backend_v11.cpp

```diff
@@ -357,34 +651,34 @@ HwComposerBackend_v11::sleepDisplay(bool sleep)
         m_vsyncTimeout.stop();
         hwc_device->eventControl(hwc_device, 0, HWC_EVENT_VSYNC, 0);
 
-#ifdef HWC_DEVICE_API_VERSION_1_4
-        if (hwc_version == HWC_DEVICE_API_VERSION_1_4) {
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->setPowerMode(hwc_device, 0, HWC_POWER_MODE_OFF));
-        } else
-#endif
-#ifdef HWC_DEVICE_API_VERSION_1_5
-        if (hwc_version == HWC_DEVICE_API_VERSION_1_5) {
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->setPowerMode(hwc_device, 0, HWC_POWER_MODE_OFF));
-        } else
-#endif
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->blank(hwc_device, 0, 1));
+        for (int i=0; i<num_displays; i++) {
+            if (g_unblanked_displays[i]) {
+                blankDisplay(i, true);
+            }
+        }
     } else {
-#ifdef HWC_DEVICE_API_VERSION_1_4
-        if (hwc_version == HWC_DEVICE_API_VERSION_1_4) {
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->setPowerMode(hwc_device, 0, HWC_POWER_MODE_NORMAL));
-        } else
-#endif
-#ifdef HWC_DEVICE_API_VERSION_1_5
-        if (hwc_version == HWC_DEVICE_API_VERSION_1_5) {
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->setPowerMode(hwc_device, 0, HWC_POWER_MODE_NORMAL));
-        } else
+#if 0
+        for (int i=0; i<num_displays; i++) {
+            blankDisplay(i, false);
+        }
 #endif
-            HWC_PLUGIN_EXPECT_ZERO(hwc_device->blank(hwc_device, 0, 0));
-
-        if (hwc_list) {
-            hwc_list->flags |= HWC_GEOMETRY_CHANGED;
+        if (g_external_connected) {
+            if (g_unblanked_displays[0]) {
+                fprintf(stderr, "Blanking internal display\n");
+                blankDisplay(0, true);
+            }
+            fprintf(stderr, "Unblanking external display\n");
+            blankDisplay(1, false);
```

Comment:
Since display 0 is assumed to be always internal display and display 1 the external display would normally avoid hard-coded magical numbers and instead introduce enumerations like InternalDisplay, ExternalDisplay passed to the functions. Now debug prints above are used to document 0=internal, 1=external